dataline-workers/src/main/java/io/dataline/workers/singer/SingerJsonStreamFactory.java

```diff
@@ -59,17 +58,13 @@ public SingerJsonStreamFactory() {
   }
 
   public Stream<SingerMessage> create(BufferedReader bufferedReader) {
-    return bufferedReader.lines().filter(singerProtocolValidator).map(this::parseJsonOrNull).filter(Objects::nonNull);
-  }
-
-  private SingerMessage parseJsonOrNull(String record) {
-    Optional<SingerMessage> message = Jsons.tryDeserialize(record, SingerMessage.class);
-    if (message.isPresent()) {
-      return message.get();
-    } else {
-      LOGGER.info("Record was not a json representation of a SingerMessage. Record: {}", record);
-      return null;
-    }
+    return bufferedReader
+        .lines()
```

Comment:
this is one of the only place where the behavior might change.

I changed it because the previous code was deserializing twice and the tryDeserialize was actually useless since the validator was doing an unguarded deserialization.